### PR TITLE
[#2092] Bump libs3 dependency and fix build hook (main)

### PR DESCRIPTION
**Pull request by alanking**
Addresses #2092

Companion PR: irods/externals#180

Still need to test this out on other platforms.

### CODE REVIEW

**Review by korydraughn — APPROVED** [review]
Looks good to me.

File: irods_consortium_continuous_integration_build_hook.py [hunk 1]
```diff
@@ -9,21 +9,53 @@
...
+        irods_python_ci_utilities.raise_not_implemented_for_distribution()
 
 def add_cmake_to_front_of_path():
     cmake_path = '/opt/irods-externals/cmake3.21.4-0/bin'
     os.environ['PATH'] = os.pathsep.join([cmake_path, os.environ['PATH']])
 
 def install_building_dependencies(externals_directory):
+    # The externals_list needs to include all dependencies, not the minimum set required for this plugin. If custom
+    # externals are being supplied via externals_directory, only the externals packages which exist in that directory
+    # will be installed.
```
[the remark this answers is not in the document]
Comment (alanking) [reply irods_consortium_continuous_integration_build_hook.py:36]: Yes, but I *think* this is the only plugin which got clobbered by yours truly. :)

[the remark this answers is not in the document]
Comment (korydraughn) [reply irods_consortium_continuous_integration_build_hook.py:36]: Sure thing. Once the release process begins, we'll know whether the other plugins need to be touched.

**Comment by alanking** [comment]
Oh, forgot to mention that I confirmed that this works on Centos 7 as well as Ubuntu 20.04. Another PR can be opened should we find kinks in other platforms

**Pull request merged**